pulp_smash/pulp3/pytest_plugin/__init__.py

```diff
@@ -506,13 +508,35 @@ def ssl_ctx_req_client_auth(
 
 
 @pytest.fixture
-def gen_object_with_cleanup():
-    obj_hrefs = []
+def add_to_cleanup():
+    """Fixture to allow pulp objects to be deleted after the test."""
+    obj_refs = []
 
+    def _add_to_cleanup(api_client, pulp_href):
+        obj_refs.insert(0, (api_client, pulp_href))
```

Comment:
Python lists are not optimized to insert at the beginning, but the collections.deque is optimized to insert from both ends. I think we should use that.

Reply:
What about calling `obj_refs.reverse()` before cleanup? Or using a reverse iterator (does python have such a thing?)?

Reply:
Then we have to reverse them? I think this is what `collections.deque` was made for actually so I think we should use it.